dace/codegen/targets/intel_fpga.py

```diff
@@ -1468,13 +1468,36 @@ def visit_Assign(self, node):
 
     def visit_BinOp(self, node):
         if node.op.__class__.__name__ == 'Pow':
+
             # Special case for integer power: do not generate dace namespaces (dace::math) but just call pow
             if not (isinstance(node.right,
                                (ast.Num, ast.Constant)) and int(node.right.n) == node.right.n and node.right.n >= 0):
+
                 left_value = cppunparse.cppunparse(self.visit(node.left), expr_semicolon=False)
-                right_value = cppunparse.cppunparse(self.visit(node.right), expr_semicolon=False)
-                updated = ast.Name(id="pow({},{})".format(left_value, right_value))
+
+                from dace.frontend.python import astutils
+                try:
+                    unparsed = symbolic.pystr_to_symbolic(
+                        astutils.evalnode(node.right, {
+                            **self.constants,
+                            'dace': dace,
+                        }))
+                    evaluated = symbolic.symstr(evaluate(unparsed, self.constants))
+                    infered_type = infer_expr_type(evaluated, self.dtypes)
+                    infered_type_str = str(infered_type)
+                    right_value = evaluated
+
+                    if infered_type_str == 'long long' or infered_type_str == 'int' or infered_type_str == 'long':
```

Comment:
Instead of comparing strings, use the returned infered_type (you should check whether this is equal to `dtypes.int32` or `dtypes.int64`)